packages/shared/src/guards.ts

```diff
@@ -126,10 +125,24 @@ export function assertArray(val: unknown): asserts val is unknown[] {
   }
 }
 
+function isTypedArray(val: unknown): boolean {
```

Comment:
Much simpler implementation than lodash's as well, since I think lodash parses the output of `toString`... Also, it will be simpler to check for bigint typed arrays when the type comes.